scene/gui/tree.cpp

```diff
@@ -2096,7 +2117,12 @@ int Tree::draw_item(const Point2i &p_pos, const Point2 &p_draw_ofs, const Size2
 				buttons_width += button_size.width + theme_cache.button_margin;
 			}
 
-			p_item->cells.write[i].text_buf->set_width(item_width);
+			int text_width = item_width;
```

Comment:
```suggestion
			int text_width = item_width - theme_cache.inner_item_margin_left - theme_cache.inner_item_margin_right;
```

This should fix the issue with the text overflow. It's already accounted for when drawing the rectangle, so we only need to add it to the text width. I've tested it and it seems to work fine.

See https://github.com/godotengine/godot/pull/75460.

Reply:
Thanks! I've noticed a new bug, but it's probably not related to this PR.

https://github.com/godotengine/godot/assets/47700418/91de2a65-b482-4a10-b313-ca9c6a05a7ac

Reply:
Can also be related to the new margin property, I guess.